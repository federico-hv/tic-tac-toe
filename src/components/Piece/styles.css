
.piece {
	width: 6em;
	height: 6em;
	cursor: pointer;
}

.border-1 {
	border: 3px solid white;
}

.border-2 {
	border: 3px solid black;
}

.piece-x {
	border: 3px solid black;
}

.square-0-0 {
	border-top: none;
	border-left: none;
}

.square-0-1 {
	border-top: none;
}

.square-0-2 {
	border-top: none;
	border-right: none;
}

.square-1-0 {
	border-left: none;	
}

.square-1-1 {

}

.square-1-2 {
	border-right: none;	
}

.square-2-0 {
	border-left: none;
	border-bottom: none;
}

.square-2-1 {
	border-bottom: none;
}

.square-2-2 {
	border-right: none;
	border-bottom: none;
}

.symbol-X,.symbol-O {
	display: flex;
	justify-content: center;
	align-items: center;
}

.symbol-X {
	color: #16ECC5;
}

.symbol-O {
	color: #EB7272;
}

.spancito {
	font-size: 5em;
}

.blink {
	color: #D0A1FF;
	animation: blinker 1.5s linear infinite;
}

@keyframes blinker {  
	50% { opacity: 0; }
}