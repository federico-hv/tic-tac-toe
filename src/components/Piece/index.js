import React from 'react';
import './styles.css';

const Piece = ({symbol, onSelect, squareIndex, player, highLight}) => (
	<div className={`piece square-${squareIndex} border-${player} symbol-${symbol}`} onClick={symbol === '' ? ()=>onSelect(symbol) : null}>
		<span className={highLight ? `spancito blink` : "spancito"}>{symbol}</span>
	</div>
);


export default Piece;