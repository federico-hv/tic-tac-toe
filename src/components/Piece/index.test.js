import React from 'react';
import { mount, shallow } from 'enzyme';
import Piece from './index';
import toJson from 'enzyme-to-json';



describe('<Piece />', () => {

	it('Should render an empty space', () => {
		const piece = shallow(<Piece symbol={{ value: null }} />)
		expect(piece.text()).toEqual('')
	});

	it('Should render an X', () => {
		const piece = shallow(<Piece symbol={'X'} />)
		expect(piece.text()).toEqual('X')
	});

	it('Matches snapshot', () => {
		const tree = mount(<Piece />);
		expect(toJson(tree)).toMatchSnapshot();
		tree.unmount();
	});
})