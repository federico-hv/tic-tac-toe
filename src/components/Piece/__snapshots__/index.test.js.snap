// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Piece /> Matches snapshot 1`] = `
<Piece>
  <div
    className="piece square-undefined border-undefined symbol-undefined"
    onClick={null}
  >
    <span
      className="spancito"
    />
  </div>
</Piece>
`;
